src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -58,29 +62,62 @@ export class FilterTimestamp extends HTMLElement {
     );
   }
 
-  onClearFilter = () => {
-    this.startTimeLabel.textContent = 'From Beginning';
-    this.stopTimeLabel.textContent = 'Up to End';
-    this.filterContainer.hidden = true;
+  onClearFilter(clearStart = true, clearStop = true) {
+    if (clearStart) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.clearStartTimeBtn.hidden = true;
+      delete this.timeStamp.start;
+    }
+
+    if (clearStop) {
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.clearStopTimeBtn.hidden = true;
+      delete this.timeStamp.stop;
+    }
+
+    if (Object.keys(this.timeStamp).length === 0) {
+      this.timeStamp = null;
+    }
+
+    if (!this.timeStamp) {
+      this.filterContainer.hidden = true;
+    }
 
-    this.timeStamp = null;
     this.dispatchFilterChange();
-  };
+  }
 
   onHiddenListener = () => {
     browser.menus.removeAll();
   };
 
+  handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.target) {
+        case this.clearFilterBtn:
+          this.onClearFilter();
+          break;
+        case this.clearStartTimeBtn:
+          this.onClearFilter(true, false);
+          break;
+        case this.clearStopTimeBtn:
+          this.onClearFilter(false, true);
+          break;
+        default:
+          break;
```

Comment:
The `default` in this `switch...case` is not really needed, this method would already turns into a no-op for events that we don't want to handle (because they are not coming from one of the event.target that we expect).

Reply:
done :-)